apps/opentelemetry_exporter/src/opentelemetry_exporter.erl

```diff
@@ -359,18 +262,155 @@ export(_, _Tab, _Resource, _State) ->
+
+signal_opt_name(otlp_endpoint, OtelSignal) ->
+    ?signal_opt_name(["otlp", OtelSignal, "endpoint"]);
+signal_opt_name(otlp_headers, OtelSignal) ->
+    ?signal_opt_name(["otlp", OtelSignal, "headers"]);
+signal_opt_name(otlp_protocol, OtelSignal) ->
+    ?signal_opt_name(["otlp", OtelSignal, "protocol"]);
+signal_opt_name(otlp_compression, OtelSignal) ->
+    ?signal_opt_name(["otlp", OtelSignal, "compression"]);
+signal_opt_name(otlp_timeout, OtelSignal) ->
+    ?signal_opt_name(["otlp", OtelSignal, "timeout"]);
+signal_opt_name(ssl_options, OtelSignal) ->
+    ?signal_opt_name([OtelSignal, "ssl_options"]);
+signal_opt_name(_, _OtelSignal) ->
+    undefined.
+
+env_opt_name(timeout_ms) ->
+    otlp_timeout;
+env_opt_name(ssl_options) ->
+    ssl_options;
+env_opt_name(PlainOptName) ->
+    list_to_existing_atom(lists:concat(["otlp_", PlainOptName])).
+
+parse_opt_val(endpoint, RawVal) ->
+    parse_endpoint(RawVal);
+parse_opt_val(headers, RawVal) ->
+    headers(RawVal);
+parse_opt_val(_OptName, Val) ->
+    Val.
+
+add_default_path(OtelSignal, #{path := Path} = Endpoint) ->
+    Endpoint#{path => filename:join(Path, maps:get(OtelSignal, ?DEFAULT_PATHS))};
+%% Invalid endpoint can be ignored here, it will be handled afterwards
+add_default_path(_OtelSignal, InvalidEndpoint) ->
+    InvalidEndpoint.
+
+init_by_proto(grpc, State) ->
+    #state{compression=Compression,
+           endpoint=Endpoint,
+           exporter_id=ExporterId,
+           headers=Headers,
+           ssl_options=SSLOptions} = State,
+    ChannelOpts = case Compression of
+                      undefined -> #{};
+                      Encoding -> #{encoding => Encoding}
+                  end,
+    ChannelOpts1 = case is_ssl(Endpoint) of
+                       true ->
+                           ChannelOpts#{gun_opts => #{transport => ssl,
+                                                      transport_opts => SSLOptions}};
+                       false ->
+                           ChannelOpts
+                   end,
+    case grpc_client_sup:create_channel_pool(ExporterId,
+                                             Endpoint,
+                                             ChannelOpts1) of
+        {ok, ChannelPid} ->
+            Metadata = headers_to_grpc_metadata(Headers),
+            {ok, State#state{channel_pid=ChannelPid,
+                             channel=ExporterId,
+                             grpc_metadata=Metadata}};
+        Error -> Error
+    end;
+init_by_proto(HTTP, State) when HTTP =:= http_protobuf; HTTP =:= http_json ->
+    HttpcProfile = start_httpc(State),
+    {ok, State#state{httpc_profile=HttpcProfile}}.
 
-grpcbox_endpoints(Endpoints) ->
-    [{scheme(Scheme), Host, Port, maps:get(ssl_options, Endpoint, [])} ||
-        #{scheme := Scheme, host := Host, port := Port} = Endpoint <- Endpoints].
+%% Only called on a parsed and recomposed URL which is guaranteed to be a string
+is_ssl([$h,$t,$t,$p,$s,$:,$/,$/ | _]) -> true;
```

Comment:
nit;

`you can do is_ssl("https://" ++ _) -> true;`

Reply:
fixed